lumen/ai/assistant.py

```diff
@@ -345,7 +345,6 @@ async def _get_agent(self, messages: list | str):
                 agent, reasoning = await self._choose_agent(messages, self.agents, return_reasoning=True)
                 step.stream(reasoning)
                 step.success_title = f"Selected {agent}"
-            messages.append({"role": "assistant", "content": reasoning})
```

Comment:
This didn't seem to mutate the messages--instead, we will just handle it on line 399.